<?xml version="1.0"?>
<launch>
    <arg name="roomba"           default="roomba1" />
    <arg name="hz"               default="10" />
    <arg name="is_pcl_tf"        default="false" />
    <arg name="is_debug"         default="true" />
    <arg name="config_path"      default="$(find point_cloud_object_color_detector)/config" />
    <arg name="color_param_path" default="$(find object_color_detector_params)/config" />

    <!-- object_color_detector_node -->
    <node pkg="point_cloud_object_color_detector" type="point_cloud_object_color_detector_node" name="point_cloud_object_color_detector_node" output="screen" >
        <!-- remap -->
        <remap from="pc_in"   to="$(arg roomba)/camera/depth_registered/points" />
        <remap from="bbox_in" to="$(arg roomba)/bounding_boxes" />
        <remap from="pc_out"  to="$(arg roomba)/target_cloud" />
        <remap from="ocd_out" to="$(arg roomba)/object_color_positions" />

        <!-- params -->
        <param name="HZ"        value="$(arg hz)" />
        <param name="IS_PCL_TF" value="$(arg is_pcl_tf)" />
        <param name="IS_DEBUG"  value="$(arg is_debug)" />

        <!-- rosparams -->
        <rosparam command="load" file="$(arg config_path)/target_objects/target_objects.yaml" />
        <rosparam command="load" file="$(arg color_param_path)/color_params/color_params.yaml" />
    </node>

    <!-- tf -->
    <arg name="is_tf" default="false" />
    <group if="$(arg is_tf)" >
        <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_publisher" args="0 0 0.668 0 0 0 1 $(arg roomba)/base_link $(arg roomba)/camera_color_optical_frame" />
    </group>

    <!-- rviz -->
    <arg name="is_rviz" default="true" />
    <group if="$(arg is_rviz)" >
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg config_path)/rviz/point_cloud_object_color_detector.rviz" />
    </group>

</launch>